fiftyone/core/odm/embedded_document.py

```diff
@@ -0,0 +1,202 @@
+"""
+Base classes for documents that back dataset contents.
+
+| Copyright 2017-2022, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import mongoengine
+from fiftyone.core.fields import DictField, EmbeddedDocumentField, ListField
+
+import fiftyone.core.utils as fou
+
+from .document import MongoEngineBaseDocument
+from .utils import get_implied_field_kwargs
+
+food = fou.lazy_import("fiftyone.core.odm.dataset")
+
+
+class BaseEmbeddedDocument(MongoEngineBaseDocument):
+    """Base class for documents that are embedded within other documents and
+    therefore are not stored in their own collection in the database.
+    """
+
+    _parent = None
```

Comment:
A `BaseEmbeddedDocument` instance has an `EmbeddedDocumentField` as its `_parent`. For propagating updates.